pkg/build/api/validation/validation.go

```diff
@@ -128,24 +136,20 @@ func validateRevision(revision *buildapi.SourceRevision) fielderrors.ValidationE
 	return allErrs
 }
 
-func validateOutput(output *buildapi.BuildOutput) fielderrors.ValidationErrorList {
+func validateBuildOutput(output *buildapi.BuildOutput) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
 
-	// TODO: make part of a generic ValidateObjectReference method upstream.
 	if output.To != nil {
 		kind, name, namespace := output.To.Kind, output.To.Name, output.To.Namespace
-		if len(kind) == 0 {
-			kind = "ImageStream"
-			output.To.Kind = kind
+		if kind != "ImageStream" && kind != "ImageStreamTag" && kind != "DockerImage" {
```

Comment:
Kind should be allowed to be empty, and if empty, it means ImageStream.  Validation might not be the right place to enforce that, but i don't think you should remove the logic entirely as you appear to have done.